### DISCUSSION TITLE
Location of eForms SDK 1.6.0 and Application Notice Form Editor Demo

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#373] [Q&A] [answered]
We have modified the application.yaml file of the Notice Form Editor Demo application to download SDK 1.6.0.

![image](https://user-images.githubusercontent.com/115476802/225269941-c0fcbf70-8772-4b1a-9164-5ab15c9f19f2.png)

![image](https://user-images.githubusercontent.com/115476802/225282195-267a527f-dce9-4fb5-8e00-0e500b343f06.png)

Then we have tried to generate a physical model with the Notice Form Editor Demo application.

![image](https://user-images.githubusercontent.com/115476802/225271247-f3931745-b519-4cbd-92ea-fd7ae4bc88db.png)

And it gives us the following error:

![image](https://user-images.githubusercontent.com/115476802/225281465-23e91e5b-5965-4179-8a1c-c9093ebcf35a.png)

`{"timestamp":"2023-03-15T10:23:27.507+00:00","status":500,"error":"Internal Server Error","trace":"java.io.FileNotFoundException: C:\\eforms-editor\\eforms-notice-editor\\eforms-sdks\\schemas\\maindoc\\UBL-PriorInformationNotice-2.3.xsd (El sistema no puede encontrar la ruta especificada)\r\n\tat java.base/java.io.FileInputStream.open0(Native Method)\r\n\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\r\n\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\r\n\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:112)\r\n\tat java.base/sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:86)\r\n\tat java.base/sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:184)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:654)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:150)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:860)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:824)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(XMLParser.java:141)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.parsers.DOMParser.parse(DOMParser.java:246)\r\n\tat java.xml/com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderImpl.parse(DocumentBuilderImpl.java:339)\r\n\tat java.xml/javax.xml.parsers.DocumentBuilder.parse(DocumentBuilder.java:206)\r\n\tat eu.europa.ted.eforms.noticeeditor.sorting.NoticeXmlTagSorter.buildDoc(NoticeXmlTagSorter.java:422)\r\n\tat eu.europa.ted.eforms.noticeeditor.sorting.NoticeXmlTagSorter.sortXml(NoticeXmlTagSorter.java:141)\r\n\tat eu.europa.ted.eforms.noticeeditor.helper.notice.PhysicalModel.buildPhysicalModel(PhysicalModel.java:232)\r\n\tat eu.europa.ted.eforms.noticeeditor.service.XmlWriteService.buildPhysicalModel(XmlWriteService.java:215)\r\n\tat eu.europa.ted.eforms.noticeeditor.service.XmlWriteService.buildPhysicalModel(XmlWriteService.java:128)\r\n\tat eu.europa.ted.eforms.noticeeditor.service.XmlWriteService.saveNoticeAsXml(XmlWriteService.java:61)\r\n\tat eu.europa.ted.eforms.noticeeditor.controller.XmlRestController.saveNotice(XmlRestController.java:34)\r\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\r\n\tat java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)\r\n\tat java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)\r\n\tat java.base/java.lang.reflect.Method.invoke(Method.java:566)\r\n\tat org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)\r\n\tat org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)\r\n\tat org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)\r\n\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:895)\r\n\tat org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:808)\r\n\tat org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)\r\n\tat org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1067)\r\n\tat org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:963)\r\n\tat org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)\r\n\tat org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)\r\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:517)\r\n\tat org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)\r\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:584)\r\n\tat org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:799)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1631)\r\n\tat org.eclipse.jetty.websocket.server.WebSocketUpgradeFilter.doFilter(WebSocketUpgradeFilter.java:292)\r\n\tat org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)\r\n\tat org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)\r\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\r\n\tat org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)\r\n\tat org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)\r\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\r\n\tat org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)\r\n\tat org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)\r\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\r\n\tat org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)\r\n\tat org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)\r\n\tat org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)\r\n\tat org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)\r\n\tat org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)\r\n\tat org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)\r\n\tat org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:600)\r\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)\r\n\tat org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1624)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)\r\n\tat org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1440)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)\r\n\tat org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)\r\n\tat org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1594)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)\r\n\tat org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1355)\r\n\tat org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)\r\n\tat org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)\r\n\tat org.eclipse.jetty.server.Server.handle(Server.java:516)\r\n\tat org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:487)\r\n\tat org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:732)\r\n\tat org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:479)\r\n\tat org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:277)\r\n\tat org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)\r\n\tat org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)\r\n\tat org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)\r\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)\r\n\tat org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)\r\n\tat java.base/java.lang.Thread.run(Thread.java:834)\r\n","message":"java.io.FileNotFoundException: C:\\eforms-editor\\eforms-notice-editor\\eforms-sdks\\schemas\\maindoc\\UBL-PriorInformationNotice-2.3.xsd (El sistema no puede encontrar la ruta especificada)","path":"/xml/notice/save/validation/none"}`

Do we have to move the 1.6.0 sdk to the location "\eforms-notice-editor\eforms-sdks\schemas\" instead of "\eforms-notice-editor\eforms-sdks\1.6"?

**Comment by rouschr** [comment]
It is looking for "C:\\eforms-editor\\eforms-notice-editor\\eforms-sdks\\schemas\\maindoc", it should be looking for schemas inside of eforms-sdks/1.6/
Maybe try this again, it is possible that when you tried the SDK 1.6 was not available for download by the downloader.
I do not have the time to try to reproduce your issue.
Personnally I have "eforms-sdks" inside of the project, at the same level as the pom.xml

**Comment by rouschr** [comment, marked as answer]
I was now able to reproduce the problem by looking into another issue.
There is a pending PR for this now, but you can already have a look into the fix in this branch:
https://github.com/OP-TED/eforms-notice-editor/tree/feature/fix-xsd-path-loading
